lib/curl/curl.nit

```diff
@@ -246,6 +249,116 @@ class CurlHTTPRequest
 	end
 end
 
+
+# Restful JSON request builder over HTTP
+#
+class CurlRestRequest
+	super CurlHTTPRequest
+	var method : nullable String
```

Comment:
Why don't you move the method selection logic to `CurlHTTPRequest`? It could be useful to non-rest/non-json requests.